dmutils/email/dm_mailchimp.py

```diff
@@ -0,0 +1,101 @@
+                "Mailchimp failed to create campaign for '{0}'".format(
+                    campaign_data.get("settings").get("title")
+                ),
+                extra={"error": str(e)}
+            )
+        return False
+
+    def set_campaign_content(self, campaign_id, content_data):
+        try:
+            return self.client.campaigns.content.update(campaign_id, content_data)
+        except RequestException as e:
+            self.logger.error(
+                "Mailchimp failed to set content for campaign id '{0}'".format(campaign_id),
+                extra={"error": str(e)}
+            )
+        return False
+
+    def send_campaign(self, campaign_id):
+        try:
+            self.client.campaigns.actions.send(campaign_id)
+            return True
+        except RequestException as e:
+            self.logger.error(
+                "Mailchimp failed to send campaign id '{0}'".format(campaign_id),
+                extra={"error": str(e)}
+            )
+        return False
+
+    def subscribe_new_email_to_list(self, list_id, email_address):
+        """ Will subscribe email address to list if they do not already exist in that list else do nothing"""
+        hashed_email = self.get_email_hash(email_address)
+        try:
+            return self.client.lists.members.create_or_update(
+                list_id,
+                hashed_email,
+                {
+                    "email_address": email_address,
+                    "status_if_new": "subscribed"
+                }
+            )
+        except RequestException as e:
+            # expected error due to user email being on Mailchimp blacklist. Don't fail but log.
+            if "looks fake or invalid, please enter a real email address." in e.response.json()["detail"]:
+                self.logger.error(
+                    "Expected error: Mailchimp failed to add user ({}) to list ({}). API error: The email address looks fake or invalid, please enter a real email address.".format(  # noqa
+                        hashed_email,
+                        list_id
+                    ),
+                    extra={"error": str(e)}
+                )
+                return True
+            self.logger.error(
+                "Mailchimp failed to add user ({}) to list ({})".format(
+                    hashed_email,
+                    list_id
+                ),
+                extra={"error": str(e)}
+            )
+            return False
+
+    def subscribe_new_emails_to_list(self, list_id, email_addresses):
+        success = True
+        for email_address in email_addresses:
+            if not self.subscribe_new_email_to_list(list_id, email_address):
+                success = False
+        return success
+
+    def get_email_addresses_from_list(self, list_id):
+        member_data = self.client.lists.members.all(list_id, get_all=True)
+        return [member["email_address"] for member in member_data["members"]]
```

Comment:
Sooo... why make this a wrapper of the mailchimp `client` instead of a subclass? The latter has the effect of taking the MailChimp client and augmenting it. The latter adds a micromanagement layer between the code that wants to use mailchimp and the client.